client-proxy/transports/http_transport.go

```diff
@@ -0,0 +1,10 @@
+package transports
+
+type HttpTransport struct {
+	OpenLineageURL    string
+	OpenLineageAPIKey string
```

Comment:
I'd really like to stop using API keys all over the place as the only auth mechanism. I think a token provider is a very simple interface to provide so that we can have support for OAuth tokens for anyone willing to write the token provider implementation.